packages/vue-i18n-routing/src/compatibles/utils.ts

```diff
@@ -104,13 +105,14 @@ export function resolveBridgeRoute(val: BridgeRoute) {
 export function resolvedRouteToObject(route: BridgeRoute): BridgeRoute {
   const r = resolveBridgeRoute(route)
 
+  const { search, hash } = parsePath(r.fullPath)
+
   const encodedPath = encodeURI(r.path)
```

Comment:
Just while we're at it: This line encodes a properly encoded route twice for me, using `useSwitchLocalePath`. Could `encodeURI` be removed here or be called under different circumstances?

The callstack goes like this:
- useSwitchLocalePath
- _useSwitchLocalePath
- switchLocalePath
- localePath
- resolveRoute
- resolvedRouteToObject ... where the route`s path is already urlencoded and will be erroneously encoded a second time here

Nuxt 3.8, @nuxtjs/i18n npm:@nuxtjs/i18n-edge@8.0.0-rc.5-28305512.c5b00d5

Any quick ideas?

Reply:
The encoding should probably moved someplace else or be handled differently, the issue of double encoding started after https://github.com/intlify/routing/pull/76, but without that change users experienced lack of encoding some of the time.

I've been looking into `vuejs/router` as to when and where encoding happens, `encodeURI` is hardly used there or its usage is at least modified, see [`encoding.ts`](https://github.com/vuejs/router/blob/main/packages/router/src/encoding.ts). The there comments refer to some cross-browser differences in encoding as well, maybe this could be why some users seem to experience encoding issues more than others?

This PR only fixes missing hash in URLs (only happens for URLs without search/query parameters), the encoding issue is a bit more complicated 😅 so will likely be a separate PR. Might end up copying some of the router code as is, as not all of it is exported, or perhaps we can make better use of the routers functions earlier in the code.

Reply:
Ah, I think there is already a ticket for what I described: https://github.com/nuxt-modules/i18n/issues/2523